dlrover/trainer/torch/flash_checkpoint/ckpt_backup.py

```diff
@@ -218,3 +213,90 @@ def _gather_owner_checkpoint(
                     ckpt_meta = meta
                     break
         return ckpt_shm_tensor, ckpt_meta
+
+
+class FullCkptBackupManager(BackupManger):
+    """
+    The node does not need to backup checkpoint if each rank has
+    the full checkpoint. The manager can select one rank which
+    has the full checkpoint to broadcast the checkpiont in the
+    shared memory to other ranks.
+    """
+
+    def __init__(self, local_rank, local_world_size) -> None:
+        self.local_rank = local_rank
+        self.local_world_size = local_world_size
+        self.node_rank = env_utils.get_node_rank()
+        self.rank = dist.get_rank()
+        self.node_num = env_utils.get_node_num()
+        self.current_device = torch.device("cpu")
+
+        self.backup_ranks = self._get_backup_ranks()
```

Comment:
For example: 
Define  ```get_backup_ranks()```  as abstract method. Then both ```FullCkptBackupManager``` and ```ShardCkptBackupManager``` implement the method by themself.

Reply:
The arguments in the `_get_backup_ranks` of these two class are different and the method is a private interface. We do not need to define it as an abstract method.